pkg/logs/launchers/file/launcher.go

```diff
@@ -120,18 +129,29 @@ func (s *Launcher) run() {
 		scanTicker.Stop()
 		close(s.done)
 	}()
-
 	for {
 		select {
 		case source := <-s.addedSources:
 			s.addSource(source)
 		case source := <-s.removedSources:
 			s.removeSource(source)
 		case <-scanTicker.C:
+
+			activeSourcesCopy := make([]*sources.LogSource, len(s.activeSources))
+			copy(activeSourcesCopy, s.activeSources)
+
+			scanTicker.Stop()
```

Comment:
Should we clear `addedSourceTailers` here? 
As I understand it, `addedSourceTailers` is used to track files that may have been scheduled during a scan. However if a scan is not ongoing and a source is scheduled, it is still added to `addedSourceTailers`. But it's also tracked in `activeSources`, so it will be considered in the next scan. 

by resetting `addedSourceTailers`, we only have to merge additional sources that were added during an active scan, and not between subsiqunt scans. 

(Maybe I am misunderstanding, I think some additional comments around `addedSourceTailers` and how it's supposed to work would be helpful)